### PR TITLE
Replace String.format use in LogSorter

### PR DESCRIPTION
**Pull request by milleruntime**

### CODE REVIEW

**Comment by DomGarguilo** [comment]
If you are going for maximum efficiency would if be faster to append to `String prefix = "part-r-"` instead of creating another variable - `pad`?

**Review by DomGarguilo — APPROVED** [review]
Looks good. I verified that it results in the same output.

**Review by DomGarguilo — APPROVED** [review]

**Review by EdColeman — COMMENTED** [review]
Was the format statement shown to be an issue?  That method is doing I/O.  Unless shown otherwise, it seems unlikely that calling format once at the start of the method would have any measurable impact on performance.

**Comment by milleruntime** [comment]
> Was the format statement shown to be an issue? That method is doing I/O. Unless shown otherwise, it seems unlikely that calling format once at the start of the method would have any measurable impact on performance.

It is probably not an issue since it is only called once but String.format() isn't great so I try to avoid it if possible. We created our only class in Accumulo just to avoid it. If I was dealing with byte arrays here, I would use it. Check out [FastFormat](https://github.com/apache/accumulo/blob/6e8787e36e5b89dbc9c5af5e2d2b35bab74b7f36/core/src/main/java/org/apache/accumulo/core/util/FastFormat.java)

File: server/tserver/src/main/java/org/apache/accumulo/tserver/log/LogSorter.java [hunk 1]
```diff
@@ -247,6 +247,26 @@ void writeBuffer(String destPath, List<Pair<LogFileKey,LogFileValue>> buffer, in
     }
   }
 
+  // faster replacement for String.format("part-r-%05d.rf", part);
+  private static final String suffix = ".rf";
+
+  private String getFilename(int part) {
+    String prefix;
+    if (part < 10) {
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/log/LogSorter.java:255]: Can part be negative? Might want to throw an exception in that case.

**Review by ctubbsii — COMMENTED** [review]
This doesn't strike me as particularly performance-critical, and the original code is much easier to read. It seems to me that any performance benefit in this change would be completely overshadowed by the IO involved for the RPCs to create the file and write its contents.

What motivated this change?

**Comment by ctubbsii** [comment]
> It is probably not an issue since it is only called once but String.format() isn't great so I try to avoid it if possible.

Yeah, the performance isn't great, but the readability, and ease-of-use, and code brevity are also factors to consider. We should avoid it when it matters, but I don't think we should universally avoid it. There are cases where it is preferable to alternatives. I think in this case, it's probably preferable to keep it the way it is.

**Comment by milleruntime** [comment]
Sounds like sacrificing readability of the one line of code isn't quite worth it in this case. Closing.

**Pull request closed without merging**